src/protocols/hii.rs

```diff
@@ -0,0 +1,1271 @@
+  pub block_type: u8,
+  // pub block_body: [u8]   // Intentionally skipped.
+}
+
+pub const GIBT_END: u8 = 0x00;
+pub const GIBT_GLYPH: u8 = 0x10;
+pub const GIBT_GLYPHS: u8 = 0x11;
+pub const GIBT_GLYPH_DEFAULT: u8 = 0x12;
+pub const GIBT_GLYPHS_DEFAULT: u8 = 0x13;
+pub const GIBT_GLYPH_VARIABILITY: u8 = 0x14;
+pub const GIBT_DUPLICATE: u8 = 0x20;
+pub const GIBT_SKIP2: u8 = 0x21;
+pub const GIBT_SKIP1: u8 = 0x22;
+pub const GIBT_DEFAULTS: u8 = 0x23;
+pub const GIBT_EXT1: u8 = 0x30;
+pub const GIBT_EXT2: u8 = 0x31;
+pub const GIBT_EXT4: u8 = 0x32;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GlyphInfo {
+  pub width: u16,
+  pub height: u16,
+  pub offset_x: i16,
+  pub offset_y: i16,
+  pub advance_x: i16,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtDefaultsBlock {
+  pub header: GlyphBlock,
+  pub cell: GlyphInfo,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtDuplicateBlock {
+  pub header: GlyphBlock,
+  pub char_value: crate::base::Char16,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GlyphGibtEndBlock {
+  pub header: GlyphBlock,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtExt1Block {
+  pub header: GlyphBlock,
+  pub block_type_2: u8,
+  pub length: u8,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtExt2Block {
+  pub header: GlyphBlock,
+  pub block_type_2: u8,
+  pub length: u16,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtExt4Block {
+  pub header: GlyphBlock,
+  pub block_type_2: u8,
+  pub length: u32,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtGlyphBlock {
+  pub header: GlyphBlock,
+  pub cell: GlyphInfo,
+  // pub bitmap_data: [u8],   // Intentionally skipped.
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtGlyphsBlock {
+  pub header: GlyphBlock,
+  pub cell: GlyphInfo,
+  pub count: u16
+  // pub bitmap_data: [u8],   // Intentionally skipped.
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct GibtGlyphDefaultBlock {
+  pub header: GlyphBlock,
+  pub bitmap_data: [u8],
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct GibtGlypshDefaultBlock {
+  pub header: GlyphBlock,
+  pub count: u16,
+  pub bitmap_data: [u8],
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtSkip2Block {
+  pub header: GlyphBlock,
+  pub skip_count: u16,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct GibtSkip1Block {
+  pub header: GlyphBlock,
+  pub skip_count: u8,
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct GibtVariabilityBlock {
+  pub header: GlyphBlock,
+  pub cell: GlyphInfo,
+  pub glyph_pack_in_bits: u8,
+  pub bitmap_data: [u8],
+}
+
+//
+// 33.3.8 Forms Package
+//
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct FormPackageHdr {
+  pub header: PackageHeader,
+  pub op_code_header: IfrOpHeader,
+  // Op-Codes Follow...
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct IfrOpHeader {
+  pub op_code: u8,
+  pub length_and_scope: u8,   // Length:7, Scope:1
```

Comment:
@dvdhrm, this was the only other bitfield I've encountered that I had questions about.